pkg/services/auth/jwt/auth.go

```diff
@@ -42,13 +47,46 @@ func (s *AuthService) Init() error {
 	if err := s.initClaimExpectations(); err != nil {
 		return err
 	}
+	if s.Cfg.JWTAuthKeyFile == "" && s.Cfg.JWTAuthJWKSetFile == "" && s.Cfg.JWTAuthJWKSetURL == "" {
+		if err := s.detectJWKSUri(); err != nil {
+			s.log.Debug("JWKS uri not detected due to", "err", err)
+		}
+	}
 	if err := s.initKeySet(); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// Auto configure JWKS Url following OIDC standards if the issuer is trusted
+// This can't work if you don't trust any issuers in JWTAuthExpectClaims
+func (s *AuthService) detectJWKSUri() error {
+	if s.expectRegistered.Issuer != "" {
```

Comment:
For this to work `iss` needs to be configured in `expect_claims`, right? Should probably be reflected in the documentation?